Conceptpower+Spring/src/main/java/edu/asu/conceptpower/servlet/lucene/impl/LuceneUtility.java

```diff
@@ -0,0 +1,442 @@
+
+        iterator = dict.getIndexWordIterator(POS.VERB);
+        numberOfWord = createDocuments(iterator, dict, writer);
+
+        numberOfIndexedWords += numberOfWord[0];
+        numberOfUnIndexedWords += numberOfWord[1];
+
+        try {
+            writer.commit();
+        } catch (IOException e) {
+            throw new LuceneException("Issues in writing document", e);
+        }
+
+        // Fetching DB4o Data
+
+        List<ConceptEntry> conceptEntriesList = (List<ConceptEntry>) databaseClient
+                .getAllElementsOfType(ConceptEntry.class);
+
+        numberOfWord = createDocumentsFromConceptEntries(conceptEntriesList);
+
+        numberOfIndexedWords += numberOfWord[0];
+        numberOfUnIndexedWords += numberOfWord[1];
+
+        try {
+            writer.commit();
+        } catch (IOException e) {
+            throw new LuceneException("Issues in writing document", e);
+        }
+
+        luceneDAO.storeValues(numberOfIndexedWords, LuceneAction.REINDEX);
+
+        if (numberOfUnIndexedWords > 0) {
+            throw new LuceneException("Indexing not done for " + numberOfUnIndexedWords);
+        }
+    }
+
+    /**
+     * This method fetches the concept power by iterating the fieldMap. The
+     * fieldMap contains the search criteria
+     */
+    public ConceptEntry[] queryIndex(Map<String, String> fieldMap, String operator)
+            throws LuceneException, IllegalAccessException {
+
+        if (operator == null) {
+            operator = "AND";
+        }
+
+        StringBuffer queryString = new StringBuffer();
+        int firstEntry = 1;
+
+        ConceptEntry con = new ConceptEntry();
+        java.lang.reflect.Field[] fields = con.getClass().getDeclaredFields();
+        for (java.lang.reflect.Field field : fields) {
+            SearchField search = field.getAnnotation(SearchField.class);
+            LuceneField luceneFieldAnnotation = field.getAnnotation(LuceneField.class);
+            if (search != null) {
+                String searchString = fieldMap.get(search.fieldName());
+                StringBuffer searchBuffer = new StringBuffer("(+");
+                if (searchString != null) {
+                    if (firstEntry != 1)
+                        queryString.append(" " + operator + " ");
+                    firstEntry++;
+                    queryString.append(luceneFieldAnnotation.lucenefieldName() + ":");
+                    searchString = searchString.split(" ")[0];
+                    searchBuffer.append(searchString);
+                    searchBuffer.append(")");
+                    queryString.append(searchBuffer.toString());
+                }
+            }
+        }
+
+        List<ConceptEntry> concepts = new ArrayList<ConceptEntry>();
+
+        try {
+            Query q = new QueryParser("", whiteSpaceAnalyzer).parse(queryString.toString());
+            DirectoryReader oldReader = reader;
+            reader = DirectoryReader.openIfChanged(oldReader);
```

Comment:
@jdamerow Yes it is required because if there was an index run after the application has started and user tries to query after the run of index, the reader needs to be reloaded. When I don't reload it points to the old values. And also reader will be reloaded only if there is any change in the index, else reader will remain the same.